pytest_fixtures/content_hosts.py

```diff
@@ -8,99 +8,78 @@
 from broker.broker import VMBroker
 
 from robottelo.config import settings
-from robottelo.constants import BROKER_RHEL77
 from robottelo.hosts import ContentHost
 
 
 @pytest.fixture
-def rhel_contenthost(request):
+def host_conf(request):
+    """A function-level fixture that returns parameters for VMBroker host deployment"""
+    conf = params = {}
+    if hasattr(request.node, 'callspec'):
+        params = [*request.node.callspec.params.values()][0]
+    conf['workflow'] = params.get('workflow', settings.content_host.deploy_workflow)
+    _rhelver = f"rhel{params.get('rhel_version', settings.content_host.default_rhel_version)}"
+    conf['rhel_ver'] = settings.content_host.hardware.get(f'{_rhelver}').release
+    conf['memory'] = params.get('memory', settings.content_host.hardware.get(f'{_rhelver}').memory)
+    conf['cores'] = params.get('cores', settings.content_host.hardware.get(f'{_rhelver}').cores)
+    return conf
+
+
+@pytest.fixture
+def rhel_contenthost(host_conf):
     """A function-level fixture that provides a content host object parametrized"""
     # Request should be parametrized through pytest_fixtures.fixture_markers
     # unpack params dict
-    workflow = request.param.get('workflow', settings.content_host.deploy_workflow)
-    rhel_version = request.param.get('rhel', settings.content_host.default_rhel_version)
-    # TODO: target_memory/cores, host type, other fields?
-    with VMBroker(
-        workflow=workflow, rhel_version=rhel_version, host_classes={'host': ContentHost}
-    ) as host:
+    with VMBroker(**host_conf, host_classes={'host': ContentHost}) as host:
         yield host
 
 
-@pytest.fixture
-def rhel7_contenthost():
-    """A function-level fixture that provides a content host object based on the rhel7 nick"""
-    with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as host:
+@pytest.fixture(params=[{'rhel_version': '7'}])
+def rhel7_contenthost(host_conf):
+    """A function-level fixture that provides a rhel7 content host object"""
+    with VMBroker(**host_conf, host_classes={'host': ContentHost}) as host:
         yield host
 
 
-@pytest.fixture
-def rhel7_contenthost_fips():
-    """A function-level fixture that provides a content host object based on the rhel7_fips nick"""
-    with VMBroker(nick='rhel7_fips', host_classes={'host': ContentHost}) as host:
+@pytest.fixture(scope="class", params=[{'rhel_version': '7'}])
+def rhel7_contenthost_class(host_conf):
+    """A fixture for use with unittest classes. Provides a Content Host object"""
+    with VMBroker(**host_conf, host_classes={'host': ContentHost}) as host:
         yield host
 
 
-@pytest.fixture(scope="module")
-def rhel7_contenthost_module():
+@pytest.fixture(scope='module', params=[{'rhel_version': '7'}])
+def rhel7_contenthost_module(host_conf):
     """A module-level fixture that provides a content host object based on the rhel7 nick"""
```

Comment:
this is no more using `nick`, update as host object as you mentioned for other doc string